Ignore DumpRenderTree when logging WebKit1 usage

ddkilzer opened the pull request:
#### 6f0fc8db1d712a755ae25127fc7c336fc9f6f5bd
<pre>
Ignore DumpRenderTree when logging WebKit1 usage
&lt;<a href="https://bugs.webkit.org/show_bug.cgi?id=297749">https://bugs.webkit.org/show_bug.cgi?id=297749</a>&gt;
&lt;<a href="https://rdar.apple.com/158812344">rdar://158812344</a>&gt;

Reviewed by Alexey Proskuryakov and Brent Fulgham.

* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
(WTF::CocoaApplication::isDumpRenderTree): Add.
(WTF::IOSApplication::isDumpRenderTree): Delete.
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.mm:
(WTF::CocoaApplication::isDumpRenderTree): Add.
- Incorporate IOSApplication::isDumpRenderTree() and add support for
  DumpRenderTree on macOS.
- Remove long-obsolete comments about numbered DumpRenderTree.app bundle
  IDs since we start one Simulator per DumpRenderTree.app now.
(WTF::CocoaApplication::shouldOSFaultLogForAppleApplicationUsingWebKit1):
- Add lambda to make criteria easier to read.
- Disable WebKit1 logging for DumpRenderTree and TestWebKitAPI.
- Decrease probability of logging from 1/100 to 1/1000.
(WTF::IOSApplication::isDumpRenderTree): Delete.
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
(-[WebAccessibilityObjectWrapper _accessibilityWebDocumentView]):
- Update to use CocoaApplication::isDumpRenderTree().
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::needsIOSDumpRenderTreeMainFrameRenderViewLayerIsAlwaysOpaqueHack const):
- Update to use CocoaApplication::isDumpRenderTree().

Canonical link: <a href="https://commits.webkit.org/299353@main">https://commits.webkit.org/299353@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f2369629253126b8f3cf1d61f8c26766e9abd6b5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118717 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38398 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29049 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124900 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70777 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d1f597d8-a409-41ec-a5d4-a64fe49a4d5d) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120595 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39094 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46980 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90100 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59623 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ef1c5476-a0e7-495b-9a5d-04c83eb4a3d3) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121670 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31146 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106437 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70606 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7bebf7ce-cb75-4280-90da-8135549ca6e2) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30204 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24550 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68559 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/110835 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100587 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24738 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127956 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/117231 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45624 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34436 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98743 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45988 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102657 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98525 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43969 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21972 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/42132 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18916 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45494 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/51172 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/145927 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44958 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37522 "Found 1 new JSC binary failure: testapi, Found 18598 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/array_sort2.js.default, ChakraCore.yaml/ChakraCore/test/Array/push2.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/bug215238.shrua-2.js.default, ChakraCore.yaml/ChakraCore/test/ControlFlow/forInPrimitive.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48304 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46644 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.mm:
```diff
@@ -369,6 +369,19 @@ static bool applicationBundleIsEqualTo(const String& bundleIdentifierString)
+#else
+    // We use a prefix match instead of strict equality since multiple instances of DumpRenderTree
+    // may be launched, where the bundle identifier of each instance has a unique suffix.
+    static bool isDumpRenderTree = applicationBundleIsEqualTo("org.webkit.DumpRenderTree"_s); // e.g. org.webkit.DumpRenderTree0
```
Are we doing this on iOS? That's a little surprising, given that each one lives in its own simulator, and there seems to be no need to differentiate them.

ddkilzer replied:
Somewhere along the way the code to do the string prefix check was removed, so `IOSApplication::isDumpRenderTree()` probably doesn't even work today if we're still doing that.

Also, because iOS DumpRenderTree is not an "Apple application" (bundle ID starts with `org.webkit.`), we could probably only compile the check on macOS.  (I wanted to guard against that in case it changed in the future, but I guess a bunch of simulated logs showing up for iOS DumpRenderTree.app might be okay in the short term?)

ddkilzer replied:
We start one Simulator per DumpRenderTree process now, so I'll remove these comments.

brentfulgham commented on Source/WebCore/platform/ios/wak/WebCoreThread.mm:
```diff
@@ -948,8 +948,13 @@ CFRunLoopRef WebThreadRunLoop(void)
-        os_fault_with_payload(OS_REASON_WEBKIT, 0, nullptr, 0, "WebThread enabled", 0);
+    if (WTF::CocoaApplication::isAppleApplication() && !((rand() * 1000) % 1000)) {
+        [&]() {
+            if (WTF::CocoaApplication::isDumpRenderTree())
```
Should we do this for TestWebKitAPI, too?

brentfulgham replied:
Maybe we should ignore MiniBrowser, too?

ddkilzer replied:
MiniBrowser doesn't need the checks because it's bundle IDs do not start with "`com.apple.`":
- MiniBrowser: `org.webkit.MiniBrowser`
- MobileMiniBrowser: `org.webkit.MobileMiniBrowser`

TestWebKitAPI does need the fix, though:
- Command-line executable:  `com.apple.WebKit.TestWebKitAPI`
- TestWebKitAPI.app:  `org.webkit.TestWebKitAPI`

I wonder if we should just change the `DumpRenderTree` and `TestWebKitAPI` bundle IDs on macOS from `com.apple.WebKit.*` to `org.webkit.WebKit.*`?  Or some other bike-sheddable, non-`com.apple.` prefix?

ddkilzer replied:
Okay, updated PR to ignore `TestWebKitAPI`.

brentfulgham reviewed (approved):
Thank you for tackling this. r=me

Commit:
Ignore DumpRenderTree when logging WebKit1 usage
<https://bugs.webkit.org/show_bug.cgi?id=297749>
<rdar://158812344>

Reviewed by Alexey Proskuryakov and Brent Fulgham.

* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
(WTF::CocoaApplication::isDumpRenderTree): Add.
(WTF::IOSApplication::isDumpRenderTree): Delete.
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.mm:
(WTF::CocoaApplication::isDumpRenderTree): Add.
- Incorporate IOSApplication::isDumpRenderTree() and add support for
  DumpRenderTree on macOS.
- Remove long-obsolete comments about numbered DumpRenderTree.app bundle
  IDs since we start one Simulator per DumpRenderTree.app now.
(WTF::CocoaApplication::shouldOSFaultLogForAppleApplicationUsingWebKit1):
- Add lambda to make criteria easier to read.
- Disable WebKit1 logging for DumpRenderTree and TestWebKitAPI.
- Decrease probability of logging from 1/100 to 1/1000.
(WTF::IOSApplication::isDumpRenderTree): Delete.
* Source/WebCore/accessibility/ios/WebAccessibilityObjectWrapperIOS.mm:
(-[WebAccessibilityObjectWrapper _accessibilityWebDocumentView]):
- Update to use CocoaApplication::isDumpRenderTree().
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::needsIOSDumpRenderTreeMainFrameRenderViewLayerIsAlwaysOpaqueHack const):
- Update to use CocoaApplication::isDumpRenderTree().

Canonical link: https://commits.webkit.org/299353@main
